haystack/templates/pipeline.py

```diff
@@ -45,61 +62,56 @@ class PipelineTemplate:
       ```python
       from haystack.components.generators import OpenAIGenerator
       from haystack.components.generators.utils import print_streaming_chunk
-      from haystack.templates import PipelineTemplate, TemplateSource, PredefinedTemplate
+      from haystack.templates import PipelineTemplate, PredefinedPipeline
 
       # Customize the pipeline with a streaming-capable generator
-      ts = TemplateSource.from_predefined(PredefinedTemplate.QA)
-      streaming_pipe = PipelineTemplate(ts).override("generator",
-                                                               OpenAIGenerator(
-                                                                   streaming_callback=print_streaming_chunk)).build()
+      streaming_pipe = (
+        PipelineTemplate.from_predefined(PredefinedTemplate.QA)
+        .override("generator", OpenAIGenerator(streaming_callback=print_streaming_chunk))
+        .build()
+      )
       streaming_pipe.run(data={"question": "What's the capital of Germany? Tell me about it"})
       ```
 
     - **Customizing for Specific Tasks**: Building a pipeline for document indexing with specific components tailored
     to the task.
       ```python
       from haystack.components.embedders import SentenceTransformersDocumentEmbedder
-      from haystack.templates import PipelineTemplate, TemplateSource, PredefinedTemplate
+      from haystack.templates import PipelineTemplate, PredefinedPipeline
 
       # Customize the pipeline for document indexing with specific components, include PDF file converter
-      ts = TemplateSource.from_predefined(PredefinedTemplate.INDEXING)
-      ptb = PipelineTemplate(ts, template_params={"use_pdf_file_converter": True})
+      ptb = PipelineTemplate.from_predefined(PredefinedTemplate.INDEXING)
       ptb.override("embedder", SentenceTransformersDocumentEmbedder(progress_bar=True))
-      pipe = ptb.build()
+      pipe = ptb.build(template_params={"use_pdf_file_converter": True})
 
-      result = pipe.run(data={
-          "sources": ["some_text_file.txt", "another_pdf_file.pdf"]})
+      result = pipe.run(data={"sources": ["some_text_file.txt", "another_pdf_file.pdf"]})
       print(result)
       ```
 
     The `PipelineTemplate` is designed to offer both ease of use for common pipeline configurations and the
     flexibility to customize and extend pipelines as required by advanced users and specific use cases.
     """
 
-    def __init__(
-        self, pipeline_type: PipelineType = PipelineType.EMPTY, template_params: Optional[Dict[str, Any]] = None
-    ):
+    def __init__(self, template_content: str):
         """
         Initialize a PipelineTemplate.
 
         :param pipeline_template: The template source to use. See `TemplateSource` for available methods to load
```

Comment:
This docstring is outdated. We just need a description for `template_content` and the others can be removed.